src/network/MissionControlTasks.h

```diff
@@ -92,11 +93,19 @@ class CameraStreamTask : public util::AsyncTask<> {
 	void task(std::unique_lock<std::mutex>& lock) override;
 
 private:
+	struct stream_data_t {
+		std::shared_ptr<video::H264Encoder> encoder;
+		std::shared_ptr<robot::types::CameraHandle> cam_handle;
```

Comment:
The camera handle is a reference to the open camera. When all references get cleaned up, the camera is closed, allowing other ones to open.